http/oidc/src/main/java/org/wildfly/security/http/oidc/AuthenticatedActionsHandler.java

```diff
@@ -0,0 +1,170 @@
+            facade.getResponse().end();
+            return true;
+        }
+        if (!deployment.isExposeToken()) {
+            facade.getResponse().setStatus(200);
+            facade.getResponse().end();
+            return true;
+        }
+        // Don't allow a CORS request if we're not validating CORS requests
+        String origin = facade.getRequest().getHeader(CorsHeaders.ORIGIN);
+        if (! deployment.isCors() && origin != null && ! origin.equals("null")) {
+            facade.getResponse().setStatus(200);
+            facade.getResponse().end();
+            return true;
+        }
+        return false;
+    }
+
+    protected boolean corsRequest()  {
+        if (! deployment.isCors()) return false;
+        OidcSecurityContext securityContext = facade.getSecurityContext();
+        String origin = facade.getRequest().getHeader(CorsHeaders.ORIGIN);
+        origin = "null".equals(origin) ? null : origin;
+        String exposeHeaders = deployment.getCorsExposedHeaders();
+
+        /*if (deployment.getPolicyEnforcer() != null) {
+            if (exposeHeaders != null) {
+                exposeHeaders += ",";
+            } else {
+                exposeHeaders = "";
+            }
+
+            exposeHeaders += "WWW-Authenticate";
+        }*/
+
+        String requestOrigin = getOrigin(facade.getRequest().getURI());
+        log.debugv("Origin: {0} uri: {1}", origin, facade.getRequest().getURI());
+        if (securityContext != null && origin != null && ! origin.equals(requestOrigin)) {
+            AccessToken token = securityContext.getToken();
+            List<String> allowedOrigins = token.getAllowedOrigins();
+
+            log.debugf("Allowed origins in token: %s", allowedOrigins);
+
+            if (allowedOrigins == null || (!allowedOrigins.contains("*") && !allowedOrigins.contains(origin))) {
+                if (allowedOrigins == null) {
+                    log.debugv("allowedOrigins was null in token");
+                } else {
+                    log.debugv("allowedOrigins did not contain origin");
+                }
+                facade.getResponse().sendError(403);
+                facade.getResponse().end();
+                return true;
+            }
+            log.debugv("returning origin: {0}", origin);
+            facade.getResponse().setStatus(200);
+            facade.getResponse().setHeader(CorsHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+            facade.getResponse().setHeader(CorsHeaders.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
+            if (exposeHeaders != null) {
+                facade.getResponse().setHeader(CorsHeaders.ACCESS_CONTROL_EXPOSE_HEADERS, exposeHeaders);
+            }
+        } else {
+            log.debugv("cors validation not needed as we are not a secure session or origin header was null: {0}", facade.getRequest().getURI());
+        }
+        return false;
+    }
+
+    /*private boolean isAuthorized() {
```

Comment:
I know you mention some clean up, I assume this is one of the areas.

Reply:
Yes, this will be cleaned up along with some other things that have been commented out.